### PR TITLE
Event updates

### PR DESCRIPTION
**Pull request by Darkhax**
This PR exposes several new events.

- LivingDeathEvent
- LivingHurtEvent
- LivingHealEvent
- XPChange
- LevelChange
- AdvancementEvent
- DestroyItemEvent
- FishItemEvent

Also exposed DamageSource's getters.

### CODE REVIEW

File: src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/ExpandPlayerDestroyItemEvent.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	/**
+	 * Gets a snapshot of the item from before it broke. Modifying this item
+	 * will have no effect and it should be treated as unmodifiable.
+	 * 
+	 * @return The original item from before it was broken.
+	 */
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("destroyedItem")
+    public static ItemStack getDestroyedItem(PlayerDestroyItemEvent internal) {
+        return internal.getOriginal();
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/ExpandPlayerDestroyItemEvent.java]: IItemStacks are a bit of a special case in terms of natives, can you change this to return an IItemStack instead (just wrap `internal.getOriginal()` in a `new MCItemStack(internal.getOriginal()`

This also solves the mutability issue as `MCItemStack` is immutable.

File: src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/xp/ExpandLevelChangedEvent.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import com.blamejared.crafttweaker_annotations.annotations.NativeTypeRegistration;
+
+import net.minecraftforge.event.entity.player.PlayerXpEvent.LevelChange;
+
+/**
+ * This event is fired when the player's level count is changed. If the amount
+ * is positive the levels are being added. If they are negative levels are
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/xp/ExpandLevelChangedEvent.java]: > If they are negative levels are

shouldn't this be:
```
If they are negative, levels are
```

First read through of this I was confused as to what `negative levels` was.

File: src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/xp/ExpandXpChangedEvent.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import com.blamejared.crafttweaker_annotations.annotations.NativeTypeRegistration;
+
+import net.minecraftforge.event.entity.player.PlayerXpEvent.XpChange;
+
+/**
+ * This event is fired when the player's xp count is changed. If the amount is
+ * positive the xp is being added. If the xp is negative the xp is being
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/player/xp/ExpandXpChangedEvent.java]: > If the xp is negative the xp is being

shouldn't this be:
```
If the xp is negative, the xp is being
```
?

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

**Comment by Darkhax** [comment]
@jaredlll08 I made the changes you asked for.

**Pull request merged**